### PR TITLE
Add support for nestled self-referenced generics

### PR DESCRIPTION
**Pull request by mikaelblomstrom**
This is a PR to resolve issue: #500

We are trying to use Crnk in a project and have some issues with crnk-core when resolving self-referential type-declarations on MappedSuperclasses used in crnk-data-jpa. Crnk cannot properly resolve MappedSuperclass of type: `class Foo<T extends Bar<T>> implements Bar<T>` for the opposite resource type.

This PR contains a correction for this, with included JUnit tests.

This PR also requires an updated TypeTools library, see PR jhalterman/typetools#53 that is pending approval. (Same update of TypeTooles is referenced in crnk-project/crnk-framework#512)

### CODE REVIEW

**Comment by mikaelblomstrom** [comment]
The JUnit tests are failing with `java.lang.StackOverflowError` due to issue in the TypeTools, as mentioned above, the PR jhalterman/typetools#53 will solve this.

**Comment by remmeier** [comment]
ok, just le me know when that is resolved

**Comment by lorenzleutgeb** [comment]
Just for reference, I PRed jhalterman/typetools#53 for exactly the same reason (we also used Crnk).

**Comment by mikaelblomstrom** [comment]
> ok, just le me know when that is resolved

The needed correction in typetools is now available in typetools-0.6.2

**Comment by mikaelblomstrom** [comment]
@remmeier the pull request is updated with the type tools 0.6.2 version, resolving the `java.lang.StackOverflowError` issues.

**Review by remmeier — APPROVED** [review]

**Comment by mikaelblomstrom** [comment]
Above was an unnecessary update. Any thing needed from me before this can be merged?

**Comment by filipneculciu** [comment]
@remmeier Could you, please, take a look at this PR and approve if it's OK? It seems to have gone unnoticed for a while.

File: crnk-core/build.gradle
Comment (lorenzleutgeb) [inline crnk-core/build.gradle:4]:
```suggestion
    compile 'net.jodah:typetools:0.6.3'
```

File: versions.properties
Comment (lorenzleutgeb) [inline versions.properties:2]:
```suggestion
typetools.version=0.6.3
```

**Review by remmeier — APPROVED** [review]

**Comment by remmeier** [comment]
could you rebase and resolve the conflucts?

**Comment by filipneculciu** [comment]
Rebased.

**Pull request merged**
